worker/worker.go

```diff
@@ -31,6 +31,8 @@ type Process struct {
 	created time.Time
 	log     *zap.Logger
 
+	maxExecs uint64
```

Comment:
Same, no need to create an additional field. Modifying existing `max_jobs` would be enough.

Reply:
The idea is to give each process (not poll `max_jobs`) its own limit (`maxExecs`). It will be exposed once upon process (worker) creation.
​
I can put this mechanism here:
```
if sp.cfg.MaxJobs != 0 && w.State().NumExecs() >= sp.cfg.MaxJobs {
```
like
`w.State().NumExecs() + generate_some_random_value() >= sp.cfg.MaxJobs`
but it's too hot for random generation.